src/rdpc/analysesProcessor.ts

```diff
@@ -0,0 +1,281 @@
+};
+
+const retryConfig = {
+  factor: 2,
+  retries: 5,
+  minTimeout: 1000,
+  maxTimeout: Infinity,
+};
+
+export const fetchSeqExpAnalyses = async (
+  studyId: string,
+  url: string,
+  from: number,
+  size: number
+): Promise<Analysis[]> => {
+  const query = buildQuery(studyId, from, size);
+
+  return await promiseRetry<Analysis[]>(async (retry) => {
+    try {
+      logger.info("Fetching sequencing experiment analyses from rdpc.....");
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify({ query }),
+        headers: {
+          "Content-type": "application/json",
+        },
+      });
+      return (await response.json()).data
+        .SequencingExperimentAnalyses as Analysis[];
+    } catch (err) {
+      logger.warn(
+        `Failed to fetch sequencing experiment analyses: ${err}, retrying...`
+      );
+      return retry(err);
+    }
+  }, retryConfig).catch((err) => {
+    logger.error(
+      `Failed to fetch sequencing experiment analyses of program: ${studyId} from RDPC ${url} after ${retryConfig.retries} attempts: ${err}`
+    );
+    throw err;
+  });
+};
+
+type StreamState = {
+  currentPage: number;
+};
+
+export const analysisStream = async function* (
+  studyId: string,
+  url: string,
+  config?: {
+    chunkSize?: number;
+    state?: StreamState;
+  }
+): AsyncGenerator<Analysis[]> {
+  const chunkSize = config?.chunkSize || 1000;
+  const streamState: StreamState = {
+    currentPage: config?.state?.currentPage || 0,
+  };
+  while (true) {
+    const page = await fetchSeqExpAnalyses(
+      studyId,
+      url,
+      streamState.currentPage,
+      chunkSize
+    );
+    streamState.currentPage = streamState.currentPage + chunkSize;
+
+    if (page && page.length > 0) {
+      yield page;
+    } else {
+      break;
+    }
+  }
+};
+
+/**
+ * Extracts donor-run relation from analysis, this is done by the following steps:
+ * Creates inputAnalyses-run map for each donor by grouping runs by inputAnalyses,
+ * creates donor-inputAnalyses-run map by determining the latest run for each inputAnalyses,
+ * aggregates donor-inputAnalyses-run map by donor id.
+ * @param analyses RDPC analysis array
+ */
+export const toDonorCentric = (
+  analyses: Analysis[]
+): RunsByAnalysesByDonors => {
+  const result = analyses.reduce<RunsByAnalysesByDonors>((acc, analysis) => {
+    const donorWithLatestRun = analysis.donors.reduce<RunsByAnalysesByDonors>(
+      (_acc, donor) => {
+        const inputAnalysesMap = _(analysis.runs)
+          .groupBy((run) =>
+            HashCode(
+              _(run.inputAnalyses)
+                .map((a) => a.analysisId)
+                .orderBy()
+                .join("-")
+            )
+          )
+          .value() as RunsByInputAnalyses;
+
+        Object.entries(inputAnalysesMap).forEach(([inputId, runs]) => {
+          const latestRun = getLatestRun(runs);
+          const run = latestRun === undefined ? [] : [latestRun];
```

Comment:
why `[latestRun]`?

Reply:
in fact if there's just gonna be 1, we can already just flatten the return of this `toDonorCentric` to be `{ [donorId: string]: Run[]}`. 
Unless you're planning something for the later analyses, so up to you to decide :)

Reply:
I will keep it for now because for the next pr